docs/src/guides/architecture-patterns.md

```diff
@@ -0,0 +1,92 @@
+
+These classes are instantiated as needed, using either an `ArcGISContext` or an `ArcGISContextManager` to provide platform and identity information. The [`ArcGISContextManager` documentation](./context.html) has additional background information.
+
+Once instantiated, you can call the exposed functions without passing in platform or identity information.
+
+```ts
+import { HubProjectManager, ArcGISContextManager } from "@esri/hub-common";
+
+// Typically your application will manage a context manager instance
+const ctxMgr = await ArcGISContextManager.create({
+  username: "casey",
+  password: "abc123",
+});
+// create a project manager
+const projectMr = HubProjectManager.init(ctxMgr);
+// get a project via it's slug
+const smithStProject = await HubProjectManager.fetch("smith-st-project");
+// update some properties
+smithStProject.state = "active";
+// use the .update method on the manager, and get a new object back
+const updated = await HubProjectManager.update(smithStProject);
+// Note: Manager's return new objects from the .update function. In order
+// for modern web frameworks to track changes in data, you will want to
+// return the result of the update function vs just the object you passed in.
+```
+
+### Working with Modern Web Frameworks
+
+All modern web frameworks (React, Angular, Vue, Ember etc) leverage some form of change tracking to determine when to re-render the UI. Typically this is done via object equality (aka `===`) or ES6 proxy objects, both of which are very efficient, but add complexity when working with Class instances. To keep things simple for all frameworks, Hub.js has enforced separation between the Manager classes and the Data. All Manager functions which make changes to the data, will return a new, updated instance of the data.
+
+### Composition via Interfaces and Functions
+
+Manager classes and the POJO's they work with are defined via interface composition. The Managers all implement the `IHubEntityManager` interface, which defines the basic CRUD+Search operations.
+
+Additionally, managers will also implement "trait manager interfaces" (i.e. `IWithLayoutManager`), which define behaviors which the manager will provide implementations for. This allows us to share both structure and behavior across many different entities, while still enabling per-entity customization where necessary.
+
+Similarly, the POJO's will conform to the entity specific interface (i.e. `IHubProject`) but those interfaces are composed with "trait interfaces" to ensure consistent properties are used across the types.
+
+For example `IHubProject` extends the `IWithSlug` iterface, and the `HubProjectManager` implements the `IWithSlugManager` interface, operating on the properties defined by `IWithSlug`.
```

Comment:
Show code example

Reply:
There is a diagram just below this section which should lay it all out 
![image](https://user-images.githubusercontent.com/119129/154295673-a39fae2c-115a-47f3-a630-eb2d970e3b53.png)